reactor-core/src/test/java/reactor/core/publisher/FluxIterableTest.java

```diff
@@ -44,11 +50,104 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.mock;
 
 public class FluxIterableTest {
 
 	final Iterable<Integer> source = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
 
+	static Stream<Function<Flux, Flux>> factory() {
+		return Stream.of(new Function<Flux, Flux>() {
+			@Override
+			public Flux apply(Flux flux) {
+				return flux;
+			}
+
+			@Override
+			public String toString() {
+				return "normal fast-path";
+			}
+		}, new Function<Flux, Flux>() {
+			@Override
+			public Flux apply(Flux flux) {
+				return flux.filter(__ -> true);
+			}
+
+			@Override
+			public String toString() {
+				return "conditional fast-path";
+			}
+		}, new Function<Flux, Flux>() {
+			@Override
+			public Flux apply(Flux flux) {
+				return flux.limitRate(1);
+			}
+
+			@Override
+			public String toString() {
+				return "fused";
+			}
+		}, new Function<Flux, Flux>() {
+			@Override
+			public Flux apply(Flux flux) {
+				return flux.hide()
+				           .limitRate(1);
+			}
+
+			@Override
+			public String toString() {
+				return "normal slow-path";
+			}
+		}, new Function<Flux, Flux>() {
+			@Override
+			public Flux apply(Flux flux) {
+				return flux.filter(__ -> true)
+				           .hide()
+				           .limitRate(1);
+			}
+
+			@Override
+			public String toString() {
+				return "conditional slow-path";
+			}
+		}, new Function<Flux, Flux>() {
+			@Override
+			public Flux apply(Flux flux) {
+				return flux.filter(__ -> true)
+				           .limitRate(1);
+			}
+
+			@Override
+			public String toString() {
+				return "conditional-fused";
+			}
+		});
+	}
+
+	@ParameterizedTestWithName
+	@MethodSource("factory")
+	public void testFluxIterableEmptyCase(Function<Flux, Flux> fn) {
+		Iterable<String> iterable = mock(Iterable.class);
+		Mockito.when(iterable.spliterator())
+		       .thenReturn(mock(Spliterator.class));
+
+		StepVerifier.create(
+				Flux.fromIterable(iterable)
+					.as(fn)
+				    .next()
+            )
+		    .expectSubscription()
+		    .expectComplete()
+		    .verify();
+	}
+
+	@Test
+	public void testFluxIterableSlowPath() {
+		Iterable<String> iterable = mock(Iterable.class);
+		Mockito.when(iterable.spliterator()).thenReturn(mock(Spliterator.class));
+		Flux.fromIterable(iterable).limitRate(1).next().block();
```

Comment:
It looks like this case is covered in the factory.

Reply:
right. duplication. fixed